attest/eventlog.go

```diff
@@ -257,18 +258,19 @@ func (a *AKPublic) validateQuote(quote Quote, pcrs []PCR, nonce []byte) error {
 
 	sigHash := a.Hash.New()
 	sigHash.Write(quote.Quote)
+	digest := sigHash.Sum(nil)
 
 	switch pub := a.Public.(type) {
 	case *rsa.PublicKey:
-		if sig.RSA == nil {
-			return fmt.Errorf("rsa public key provided for ec signature")
-		}
 		sigBytes := []byte(sig.RSA.Signature)
-		if err := rsa.VerifyPKCS1v15(pub, a.Hash, sigHash.Sum(nil), sigBytes); err != nil {
+		if err := rsa.VerifyPKCS1v15(pub, a.Hash, digest, sigBytes); err != nil {
+			return fmt.Errorf("invalid quote signature: %v", err)
+		}
+	case *ecdsa.PublicKey:
+		if !ecdsa.Verify(pub, digest, sig.ECC.R, sig.ECC.S) {
 			return fmt.Errorf("invalid quote signature: %v", err)
```

Comment:
There's no err here to report; we get a `bool` back from Verify. Just drop the ": %v"